pennylane/qaoa/cycle.py

```diff
@@ -685,8 +686,8 @@ def _inner_net_flow_constraint_hamiltonian(
         ops.append(qml.Z(wires))
 
     coeffs, ops = _square_hamiltonian_terms(coeffs, ops)
-    H = Hamiltonian(coeffs, ops)
-    H.simplify()
+    H = qml.Hamiltonian(coeffs, ops)
+    H = H.simplify()
```

Comment:
Has `Hamiltonian.simplify()` been updated to not mutate in-place? If not, then I suggest leaving it as is.
```suggestion
    H.simplify()
```

Reply:
So `Hamiltonian.simplify()` mutates in-place but also returns `self`, whereas `LinearCombination.simplify()` returns a new `LinearCombination`. This change would work with both without having to explicitly check whether `H` is a `Hamiltonian` or a `LinearCombination`.

Reply:
Yep

Reply:
Thanks for clarifying.